trade-finance-manager-ui/server/controllers/case/cancellation/reason-for-cancelling.controller.ts

```diff
@@ -14,17 +16,34 @@ export type PostReasonForCancellingRequest = CustomExpressRequest<{ params: { _i
  * @param req - The express request
  * @param res - The express response
  */
-export const getReasonForCancelling = (req: GetReasonForCancellingRequest, res: Response) => {
+export const getReasonForCancelling = async (req: GetReasonForCancellingRequest, res: Response) => {
   const { _id } = req.params;
-  const { user } = asUserSession(req.session);
-
-  const reasonForCancellingViewModel: ReasonForCancellingViewModel = {
-    activePrimaryNavigation: PRIMARY_NAVIGATION_KEYS.ALL_DEALS,
-    user,
-    ukefDealId: '0040613574', // TODO: DTFS2-7350 get values from database
-    dealId: _id,
-  };
-  return res.render('case/cancellation/reason-for-cancelling.njk', reasonForCancellingViewModel);
+  const { user, userToken } = asUserSession(req.session);
+  try {
+    const deal = await api.getDeal(_id, userToken);
+
+    if (!deal || 'status' in deal) {
```

Comment:
```suggestion
    if (!deal) {
```

Reply:
`getDeal` returns a really weird object if the deal id isnt valid so I'm accounting for that here.

Reply:
Can it be accomplished using optional chaining please?

Reply:
This is checking for the status though; I'm not seeing this in other areas?

Reply:
I think this is why we were originally keen to `throw` errors to be handled at this layer - without `throw`ing the error and returning objects like this, we have to check if the returned object has a `status` field to know it's errored, instead of just catching an error (and I think as we spotted live on call with Alex, quite a few places currently don't handle this correctly when they're returning this arbitrary `status` object).

Reply:
Is `!deal?.status` not an option?

Reply:
`!deal?.status` would be different: 

Suppose we have an error object returned that is `{status: 400, data: 'abc'}`

**Then with the current implementation: `if (!deal || 'status' in deal) {`**
`status` is in `{status: 400, data: 'abc'}` so this is `true` and this `if` block runs. 

**With the suggestion: `if (!deal?.status) {`**
`deal?.status` is `true` so `!deal?.status` is `false` so this block would not run. 

So these are two different behaviours. I don't think we can make Mark's implementation more succinct.